ocp_utilities/operators.py

```diff
@@ -0,0 +1,172 @@
+            f"{pformat(subscription)}"
+        )
+        raise
+
+
+def wait_for_operator_install(admin_client, subscription, timeout=TIMEOUT_5MIN):
+    """
+    Wait for the operator to be installed, including InstallPlan and CSV ready.
+
+    Args:
+        admin_client (DynamicClient): Cluster client.
+        subscription (Subscription): Subscription instance.
+        timeout (int): Timeout in seconds to wait for operator to be installed.
+    """
+    install_plan = wait_for_install_plan_from_subscription(
+        admin_client=admin_client, subscription=subscription
+    )
+    install_plan.wait_for_status(status=install_plan.Status.COMPLETE, timeout=timeout)
+    wait_for_csv_successful_state(
+        admin_client=admin_client,
+        subscription=subscription,
+    )
+
+
+def wait_for_csv_successful_state(admin_client, subscription, timeout=TIMEOUT_10MIN):
+    """
+    Wait for CSV to be ready.
+
+    Args:
+        admin_client (DynamicClient): Cluster client.
+        subscription (Subscription): Subscription instance.
+        timeout (int): Timeout in seconds to wait for CSV to be ready.
+    """
+    csv = get_csv_by_name(
+        csv_name=subscription.instance.status.installedCSV,
+        admin_client=admin_client,
+        namespace=subscription.namespace,
+    )
+    csv.wait_for_status(status=csv.Status.SUCCEEDED, timeout=timeout)
+
+
+def get_csv_by_name(admin_client, csv_name, namespace):
+    """
+    Gets CSV from a given namespace by name
+
+    Args:
+        admin_client (DynamicClient): Cluster client.
+        csv_name (str): Name of the CSV.
+        namespace (str): namespace name.
+
+    Returns:
+        ClusterServiceVersion: CSV instance.
+
+    Raises:
+        NotFoundError: when a given CSV is not found in a given namespace
+    """
+    csv = cluster_resource(ClusterServiceVersion)(
+        client=admin_client, namespace=namespace, name=csv_name
+    )
+    if csv.exists:
+        return csv
+    raise ResourceNotFoundError(f"CSV {csv_name} not found in namespace: {namespace}")
+
+
+def install_operator(
+    admin_client, name, channel, source, target_namespaces, timeout=TIMEOUT_30MIN
+):
+    """
+    Install operator on cluster.
+
+    Args:
+        admin_client (DynamicClient): Cluster client.
+        name (str): Name of the operator to install.
+        channel (str): Channel to install operator from.
+        source (str): CatalogSource name.
+        target_namespaces (list): Target namespaces for the operator install.
+        timeout (int): Timeout in seconds to wait for operator to be ready.
+    """
+
+    if target_namespaces:
+        for namespace in target_namespaces:
+            ns = Namespace(client=admin_client, name=namespace)
+            if ns.exists:
+                continue
+
+            ns.deploy(wait=True)
+
+    else:
+        ns = Namespace(client=admin_client, name=name)
+        if not ns.exists:
+            ns.deploy(wait=True)
+
+    OperatorGroup(
```

Comment:
if we have multiple namespaces, should OperatorGroup and subscription be created in each ns?

Reply:
No, AFAIK the namespace is only to get permissions to the operator on them